test/automated-suite.js

```diff
@@ -0,0 +1,212 @@
+
+describe('Test SW-Toolbox', () => {
+  // Driver is initialised to null to handle scenarios
+  // where the desired browser isn't installed / fails to load
+  // Null allows afterEach a safe way to skip quiting the driver
+  let driver = null;
+
+  afterEach(done => {
+    if (driver === null) {
+      return done();
+    }
+
+    driver.quit()
+    .then(() => {
+      driver = null;
+      done();
+    })
+    .thenCatch(() => {
+      driver = null;
+      done();
+    });
+  });
+
+  let checkFileExists = path => {
+    return new Promise((resolve, reject) => {
+      fs.stat(path, err => {
+        if (err) {
+          return reject(err);
+        }
+        resolve();
+      });
+    });
+  };
+
+  let performTests = (browserName, driver) => {
+    // The driver methods are wrapped in a new promise because the
+    // selenium-webdriver API seems to using some custom promise
+    // implementation that has slight behaviour differences.
+    return new Promise((resolve, reject) => {
+      driver.get('http://localhost:8888/test/')
+      .then(() => {
+        return driver.executeScript('return window.navigator.userAgent;');
+      })
+      .then(userAgent => {
+        // This is just to help with debugging so we can get the browser version
+        console.log('    Browser User Agent [' + browserName + ']: ' + userAgent);
+      })
+      .then(() => {
+        // We get webdriver to wait until window.swtoolbox.testResults is defined.
+        // This is set in the in browser mocha tests when the tests have finished
+        // successfully
+        return driver.wait(function() {
+          return driver.executeScript('return ((typeof window.swtoolbox !== \'undefined\') && window.swtoolbox.testResults !== \'undefined\');');
+        });
+      })
+      .then(() => {
+        // This simply retrieves the test results from the inbrowser mocha tests
+        return driver.executeScript('return window.swtoolbox.testResults;');
+      })
+      .then(testResults => {
+        // Resolve the outer promise to get out of the webdriver promise chain
+        resolve(testResults);
+      })
+      .thenCatch(function(err) {
+        reject(err);
+      });
+    })
+    .then(testResults => {
+      if (testResults.failed.length > 0) {
+        var failedTests = testResults.failed;
+        var errorMessage = 'Issues in ' + browserName + '.\n\n' + browserName + ' had ' + testResults.failed.length + ' test failures.\n';
+        errorMessage += '------------------------------------------------\n';
+        errorMessage += failedTests.map((failedTest, i) => {
+          return `[Failed Test ${i + 1}]\n    ${failedTest.title}\n`;
+        }).join('\n');
+        errorMessage += '------------------------------------------------\n';
+        throw new Error(errorMessage);
+      }
+    });
+  };
+
+  it('should pass all tests in Chrome Stable', done => {
+    checkFileExists('/usr/bin/google-chrome-stable')
```

Comment:
FYI, you can use the npm module `which` to get the path of a binary that is in the `PATH`, and therefore check that it exists. Assuming that everything gets installed in the `PATH` on Travis this should make things work a little bit more cross platform.

Fine as a follow-up though, as getting things working on Travis is the goal here.